x-pack/plugins/security_solution/server/usage/queries/utils/get_search_for_all_rules.test.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { getSearchForAllRules } from './get_search_for_all_rules';
+
+describe('get_search_for_all_rules', () => {
+  test('it returns query merged with an aggregation sent in', () => {
+    const result = getSearchForAllRules({
+      eventLogIndex: 'test-123',
+      aggs: {
+        eventActionStatusChange: {
+          filter: {
+            term: {
+              'event.action': 'status-change',
+            },
+          },
+        },
+      },
+    });
+    expect(result).toEqual<SearchRequest>({
+      index: 'test-123',
+      size: 0,
+      track_total_hits: false,
+      aggs: {
+        eventActionStatusChange: {
+          filter: {
+            term: {
+              'event.action': 'status-change',
+            },
+          },
+        },
+      },
+      query: {
+        bool: {
+          filter: {
+            bool: {
+              must: [
+                {
+                  range: {
+                    '@timestamp': {
+                      gte: 'now-24h',
+                      lte: 'now',
```

Comment:
When are the telemetry queries ran? Is there any guarantee to consistency? In other words, can there be drift resulting in the queries running at different times than they usually run as these relative datetimes will bubble that drift into the telemetry. Would be nice if a telemetry service could push these windows down for use?

Reply:
Resolving as brought up below: https://github.com/elastic/kibana/pull/128216/files#r833090911